tools/Utilities/EnvVariablesUtility/EnvironmentVariablesAppMainWindow.xaml.cs

```diff
@@ -0,0 +1,107 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+using System;
+using System.IO;
+using System.Runtime.InteropServices;
+using DevHome.EnvironmentVariables.Win32;
+using DevHome.Telemetry;
+using EnvironmentVariablesUILib;
+using EnvironmentVariablesUILib.Helpers;
+using EnvironmentVariablesUILib.ViewModels;
+using Microsoft.UI.Dispatching;
+using Microsoft.UI.Xaml;
+using Microsoft.UI.Xaml.Controls;
+using Serilog;
+using WinUIEx;
+
+namespace DevHome.EnvironmentVariables;
+
+public sealed partial class EnvironmentVariablesMainWindow : WindowEx
+{
+    private EnvironmentVariablesMainPage EnvVariablesUtilityMainPage { get; }
+
+    private string UtilityTitle { get; set; }
+
+    private readonly Guid activityId;
+
+    private readonly Serilog.ILogger _log = Log.ForContext("SourceContext", nameof(EnvironmentVariablesApp));
+
+    public EnvironmentVariablesMainWindow(Guid activityId)
+    {
+        this.activityId = activityId;
+        this.InitializeComponent();
+
+        Activated += MainWindow_Activated;
+
+        ExtendsContentIntoTitleBar = true;
+        SetTitleBar(AppTitleBar);
+
+        var loader = new Microsoft.Windows.ApplicationModel.Resources.ResourceLoader("PowerToys.EnvironmentVariablesUILib.pri", "PowerToys.EnvironmentVariablesUILib/Resources");
+        var title = EnvironmentVariablesApp.GetService<IElevationHelper>().IsElevated ? loader.GetString("WindowAdminTitle") : loader.GetString("WindowTitle");
+        UtilityTitle = title;
+
+        Title = UtilityTitle;
+        AppWindow.SetIcon(Path.Combine(AppContext.BaseDirectory, "Assets/EnvironmentVariables/EnvironmentVariables.ico"));
+
+        var handle = this.GetWindowHandle();
+        RegisterWindow(handle);
+
+        EnvVariablesUtilityMainPage = EnvironmentVariablesApp.GetService<EnvironmentVariablesMainPage>();
+
+        TelemetryFactory.Get<ITelemetry>().Log("EnvironmentVariablesApp_EnvironmentVariablesMainWindow_Initialized", LogLevel.Measure, new EmptyEvent(), this.activityId);
+        _log.Information("EnvironmentVariablesApp MainWindow Intialized");
+    }
+
+    private void Grid_Loaded(object sender, RoutedEventArgs e)
+    {
+        MainGrid.Children.Add(EnvVariablesUtilityMainPage);
+        Grid.SetRow(EnvVariablesUtilityMainPage, 1);
+
+        TelemetryFactory.Get<ITelemetry>().Log("EnvironmentVariablesApp_EnvironmentVariablesMainWindow_GridLoaded", LogLevel.Measure, new EmptyEvent(), this.activityId);

```

Comment:
Just a note here that #2809 will change EmptyEvent() to require a privacy tags argument. This sort of event may be ProductAndServicePerformance data, same as the one above. 

There will very likely be a merge conflict between these two PRs depending on which goes in first.

Reply:
Replaced EmptyEvents with specific events tagged with PSP